### DISCUSSION TITLE
Webpack build fails during SSR build- Angular 12+

### DISCUSSION THREAD

**Discussion by hp1493** [webpack/webpack#16884] [Q&A]
# Bug report

https://stackoverflow.com/questions/75799158/webpack-hangs-during-ssr-build-angular-12

**What is the current behavior?**
when running webpack to build a server side rendering for Angular 12+ application (upgraded from angular v6) , facing issue as below  when trying to build `webpack --config webpack.server.config.js --progress`

`/Users/node_modules/webpack/lib/stats/DefaultStatsPrinterPlugin.js:1248
                                return match.replace(content, format(content));
                                             ^

RangeError: Invalid string length
    at String.replace (<anonymous>)
    at /Users//node_modules/webpack/lib/stats/DefaultStatsPrinterPlugin.js:1248:18
    at String.replace (<anonymous>)
    at Object.formatError (/Users//node_modules/webpack/lib/stats/DefaultStatsPrinterPlugin.js:1247:22)
    at context.<computed> (/Users//node_modules/webpack/lib/stats/DefaultStatsPrinterPlugin.js:1323:35)
    at Object.error.message (/Users//node_modules/webpack/lib/stats/DefaultStatsPrinterPlugin.js:590:48)
    at /Users//node_modules/webpack/lib/stats/DefaultStatsPrinterPlugin.js:1332:29
    at Hook.eval [as call] (eval at create (/Users//node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:7:16)
    at Hook.CALL_DELEGATE [as _call] (/Users//node_modules/tapable/lib/Hook.js:14:14)
    at /Users//node_modules/webpack/lib/stats/StatsPrinter.js:183:9`
 
**If the current behavior is a bug, please provide the steps to reproduce.**

1. Run  ` webpack --config webpack.server.config.js --progress` with entry as server.ts created using the guide 'https://medium.com/upstate-interactive/how-to-build-an-angular-app-with-server-side-rendering-f7f415b851c8'
**webpack server.config.ts** 
`const path = require('path');
const webpack = require('webpack');

module.exports = {
  mode: 'none',
  entry: { server: './server.ts' },
  
  externals: {
   './dist/server/main': 'require("./server/main")'
  },
  
  target: 'node',
  resolve: { extensions: ['.ts','.tsx', '.js'] },

  optimization: {
    minimize: false
  },
  output: {
    // Puts the output at the root of the dist folder
    path: path.join(__dirname, 'dist'),
    // Export a UMD of the webpacked server.ts & dependencies for rendering in Cloud Functions
    library: 'app',
    libraryTarget: 'umd',
    filename: '[name].js',
  },
  module: {
    noParse: /polyfills-.*\.js/,
    rules: [
      { test: /\.tsx?$/, loader: "ts-loader" },
      {
        // Mark files inside `@angular/core` as using SystemJS style dynamic imports.
        // Removing this will cause deprecation warnings to appear.
        test: /(\\|\/)@angular(\\|\/)core(\\|\/).+\.js$/,
        parser: { system: true },
      },
    ]
  },
  plugins: [
    new webpack.ContextReplacementPlugin(
      // fixes WARNING Critical dependency: the request of a dependency is an expression
      /(.+)?angular(\\|\/)core(.+)?/,
      path.join(__dirname, 'src'), // location of your src
      {} // a map of your routes
    ),
    new webpack.ContextReplacementPlugin(
      // fixes WARNING Critical dependency: the request of a dependency is an expression
      /(.+)?express(\\|\/)(.+)?/,
      path.join(__dirname, 'src'),
      {}
    )
  ]
}`

**What is the expected behavior?**
build should succeed without any errors from webpack module

**Other relevant information:**
webpack version:  5.76.3
Node.js version: 16.19.1
Operating System: MAC OS
Additional tools: Anguar 13 with SSR build , TS-loader : 9.2.6 , webpack-cli - 4.10.0

**Comment by alexander-akait** [comment]
Do you have additional tools like angular-cli/etc?, How you run webpack? Please avoid copy/paste your code in the issue template - create a github repo with example of the problem, sorry, it is not readable and don't provide enough context to help you

**Reply by alexander-akait** [reply]
And also please run `npm ls webpack`

**Reply by hp1493** [reply]
├─┬ @angular-devkit/build-angular@14.2.11
│ ├─┬ @angular-devkit/build-webpack@0.1402.11
│ │ └── webpack@5.76.1 deduped
│ ├─┬ @ngtools/webpack@14.2.11
│ │ └── webpack@5.76.1 deduped
│ ├─┬ babel-loader@8.2.5
│ │ └── webpack@4.46.0 deduped
│ ├─┬ copy-webpack-plugin@11.0.0
│ │ └── webpack@5.76.1 deduped
│ ├─┬ css-loader@6.7.1
│ │ └── webpack@5.76.1 deduped
│ ├─┬ less-loader@11.0.0
│ │ └── webpack@5.76.1 deduped
│ ├─┬ mini-css-extract-plugin@2.6.1
│ │ └── webpack@5.76.1 deduped
│ ├─┬ postcss-loader@7.0.1
│ │ └── webpack@5.76.1 deduped
│ ├─┬ sass-loader@13.0.2
│ │ └── webpack@5.76.1 deduped
│ ├─┬ source-map-loader@4.0.0
│ │ └── webpack@5.76.1 deduped
│ ├─┬ stylus-loader@7.0.0
│ │ └── webpack@5.76.1 deduped
│ ├─┬ webpack-dev-middleware@5.3.3
│ │ └── webpack@4.46.0 deduped
│ ├─┬ webpack-dev-server@4.11.0
│ │ └── webpack@4.46.0 deduped
│ ├─┬ webpack-subresource-integrity@5.1.0
│     └── webpack@5.76.1 deduped
└─┬ webpack-cli@3.3.11
  └─┬ webpack@4.46.0
    └─┬ terser-webpack-plugin@1.4.5
      └── webpack@4.46.0 deduped

**Reply by hp1493** [reply]
I'm unable to host source code due to compliance issue , hence for minimal reproduction linking the guide which i followed to setup this process (https://medium.com/upstate-interactive/how-to-build-an-angular-app-with-server-side-rendering-f7f415b851c8)
1. Setup server.ts ( https://gist.github.com/kseniya292/630b410db844d70e1e05e133da524ba8#file-server-ts ) 
2. create a webpack.server.config.ts for bundling the server.ts as node server bundle (https://gist.github.com/kseniya292/258af899ecc789d39c6638dc5aa9a693#file-webpack-server-config-js)
3. Run `webpack   --config webpack.server.config.js --progress` to build using webpack
4. above build fails with error in webpack  as mentioned ,
pls let me know if other information is required

**Reply by alexander-akait** [reply]
As you can see you have webpack v4, you shoukd not mix v4 and v5, because some plugins can be broken, please update all deps to the latest version, especial webpack-cli

**Reply by hp1493** [reply]
Will give it a try ! 👍

**Reply by ajay92x** [reply]
Hello @hp1493 
I'm also facing the similar problem. Did you find any solution of this problem.

**Comment by hp1493** [comment]
@ajay, couldn't solve for version 12 but problem got resolved automatically
in Angular 15
